edge/pkg/edged/edged.go

```diff
@@ -135,14 +136,18 @@ func (e *edged) Start() {
 		}
 	}
 
+	klErrChan := make(chan error, 1)
 	go func() {
 		err := DefaultRunLiteKubelet(e.context, e.KubeletServer, e.KubeletDeps, e.FeatureGate)
 		if err != nil {
-			klog.Errorf("Start edged failed, err: %v", err)
-			os.Exit(1)
+			if !kefeatures.DefaultFeatureGate.Enabled(kefeatures.ModuleRestart) {
+				klog.Errorf("Start edged failed, err: %v", err)
+				os.Exit(1)
+			}
+			klErrChan <- err
 		}
 	}()
-	e.syncPod(e.KubeletDeps.PodConfig)
+	e.syncPod(e.KubeletDeps.PodConfig, klErrChan)
```

Comment:
If run `DefaultRunLiteKubelet` failed, why not return before `e.syncPod` directly？

Reply:
Maybe we can move the pre-condition of `edged.syncPod` to `edged.Start` directly